dev/bots/prepare_package.dart

```diff
@@ -426,6 +428,47 @@ class ArchiveCreator {
     await _unzipArchive(gitFile, workingDirectory: minGitPath);
   }
 
+  /// Downloads an archive of every package that is present in the temporary
+  /// pub-cache from pub.dev. Stores the archives in
+  /// $flutterRoot/.pub-preload-cache.
+  ///
+  /// These archives will be installed in the user-level cache on first
+  /// following flutter command that accesses the cache.
+  ///
+  /// This assumes that all packages currently in the cache are installed from
+  /// pub.dev.
+  Future<void> _downloadPubPackageArchives() async {
+    final Pool pool = Pool(10); // Number of simultaneous downloads.
+    final http.Client client = http_retry.RetryClient(http.Client());
+    /// Fetch a single pacla
+    Future<void> fetchPackageArchive(String name, String version) async {
+      pool.withResource(() async {
+        final Uri url = Uri.parse('https://pub.dev/packages/$name/versions/$version.tar.gz');
```

Comment:
We could use the [supported API](https://github.com/dart-lang/pub/blob/master/doc/repository-spec-v2.md#list-all-versions-of-a-package), instead of this API end-point which is deprecated.

It's probably also a good idea to run the downloaded bytes through sha256 and compare to the version listing `archive_sha256`, so we don't risk shipping a flutter release with a half-downloaded tarball.

Reply:
Done